lightning/src/ln/chan_utils.rs

```diff
@@ -858,6 +858,210 @@ pub fn build_anchor_input_witness(funding_key: &PublicKey, funding_sig: &Signatu
+	}
+}
+
+/// Stores info about zero or one funding transaction:
+/// - none in case of unfunded channels
+///   1 in case of funded channels (can be uncofirmed or confirmed)
+/// It is prepared to handle multiple funding transaction candidates (e.g. in case of Splicing)
+pub(crate) struct ChannelFundingTransactions {
+	tx: Option<ChannelFundingTransaction>,
+}
+
+impl ChannelFundingTransactions {
+	pub(crate) fn new_empty() -> Self {
+		Self { tx: None }
+	}
+
+	pub(crate) fn is_some(&self) -> bool {
+		self.tx.is_some()
+	}
+
+	pub(crate) fn set_tx(&mut self, tx: Transaction) {
+		if let Some(cft) = self.tx.as_mut() {
+			cft.set_tx(tx);
+		} else {
+			let mut cft = ChannelFundingTransaction::new_empty();
+			cft.set_tx(tx);
+			self.tx = Some(cft);
+		}
+	}
+
+	pub(crate) fn set_txo(&mut self, txo: chain::transaction::OutPoint) {
+		if let Some(cft) = self.tx.as_mut() {
+			cft.set_txo(txo);
+		} else {
+			let mut cft = ChannelFundingTransaction::new_empty();
+			cft.set_txo(txo);
+			self.tx = Some(cft);
+		}
+	}
+
+	pub(crate) fn is_confirmed(&self) -> bool {
+		self.tx.as_ref().map_or(false, |tx| tx.is_confirmed())
+		// match &self.tx {
+		// 	None => false,
+		// 	Some(tx) => tx.is_confirmed(),
+		// }
+	}
+
+	pub(crate) fn is_unconfirmed(&self) -> bool {
+		self.tx.as_ref().map_or(true, |tx| !tx.is_confirmed())
+	}
+
+	pub(crate) fn set_confirmed(&mut self, block_hash: BlockHash, block_height: u32) {
+		if let Some(tx) = self.tx.as_mut() {
+			tx.set_confirmed(block_hash, block_height);
+		} else {
+			panic!("Set_confirmed() invoked, but there is no funding tx!");
+		}
+	}
+
+	pub(crate) fn get_tx(&self) -> Option<Transaction> {
+		match &self.tx {
+			None => None,
+			Some(tx) => match tx.get_tx() {
+				None => None,
+				Some(tx) => Some(tx.clone()),
+			}
+		}
+	}
+
+	pub(crate) fn tx_take(&mut self) -> Option<Transaction> {
+		match self.tx.take() {
+			None => None,
+			Some(tx) => match tx.get_tx() {
+				None => None,
+				Some(tx) => Some(tx.clone())
+			}
+		}
+	}
+
+	/// Returns the block hash in the funding transaction was confirmed.
+	pub(crate) fn get_tx_confirmed_in(&self) -> Option<BlockHash> {
+		match &self.tx {
+			None => None,
+			Some(tx) => tx.get_tx_confirmed_in(),
+		}
+	}
+
+	pub(crate) fn get_tx_confirmation_height(&self) -> Option<u32> {
+		self.tx.as_ref().map_or(None, |tx| tx.get_tx_confirmation_height())
+	}
+
+	/// Returns the current number of confirmations
+	pub(crate) fn get_tx_confirmations(&mut self, height: u32) -> Option<i64> {
+		self.tx.as_mut().map_or(None, |tx| tx.get_tx_confirmations(height))
+	}
+
+	/// Returns the current number of confirmations
+	pub(crate) fn get_tx_confirmations_nocheck(&self, height: u32) -> Option<u32> {
+		self.tx.as_ref().map_or(None, |tx| tx.get_tx_confirmations_nocheck(height))
+	}
+}
+
+impl Writeable for ChannelFundingTransactions {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		self.tx.write(writer)?;
```

Comment:
If the intention is to later store more than one transaction, then we should define the serialization format to account for that. Though note backwards compatibility concerns mentioned earlier.